### DISCUSSION TITLE
Conexión Git GitHub y VS Code

### DISCUSSION THREAD

**Discussion by Dancgz** [Launch-X-Latam/MisionFrontEnd#68] [Q&A]
Hola, compañerxs! 

Tengo la duda de cómo instalo Git para que esté conectada con GitHub? Y además, ya me he instalado el VS Code pero en los settings me aparece de opción que vincule con Git, esto cómo funciona para GitHub? Es rentable hacer vincular Git con VSCode?

**Comment by JosafatJimenezB** [comment]
holi, pues es de gran ayuda saber utlizar git para sacarle el maximo provecho, te dejo este tutorial de como conectar tu trabajo en local para subirlo a github usando git [ link tutorial](https://juancadh.medium.com/conectar-carpeta-local-con-repositorio-de-github-8d983798998e)
tambien te pedira autenticarte colocando tu usuario de github y usando un token (otro tuto para iniciar sesion) [tutorial token](https://docs.github.com/es/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token)

> Nota
> El token guardalo en un lugar seguro por si lo vuelves a ocupar

**Comment by Cerecero** [comment]
Tener git instalado y el github vinculado con el vscode, es demasiado util, puedes hacer pull, commit and push desde la terminal de VScode. Te facilitara la vida, el poder hacer todo lo que necesitas desde la terminal sin necesidad de usar el sitio web de github, al trabajar en tus proyectos.

Git provee control de versiones, es decir si existe algun cambio queda registro que se cambio, cuando etc...
Github es el proveedor de hosting del codigo, que te deja administrar los repositorios de git. Existen otros como Gitlab o bitbucket.

Si sabes ingles te recomiendo estos tutoriales de The Odin Project para configurar vscode y github. [Setting Up Git](https://www.theodinproject.com/lessons/foundations-setting-up-git) Y [Git Basics](https://www.theodinproject.com/lessons/foundations-git-basics#introduction), si no mandame mensaje y te ayudo a configurar.

**Reply by JosafatJimenezB** [reply]
Ese tutorial esta mejor, nice

**Reply by Dancgz** [reply]
Muchas gracias! muy útiles los tutoriales

**Comment by SilvyeCG** [comment]
Yo lo hice a través de una SSH Key

**Comment by eldhaim** [comment]
Si ya tienes git instalado solo abres una nueva carpeta en el explorador de archivos en la ruta de tu preferencia y haces esto:
1. click derecho dentro de la carpeta y seleccionas git bash 
![image](https://user-images.githubusercontent.com/86509650/192933198-5fea1ae4-af6c-496c-871e-6b3a9ed0d224.png)

2. escribes: git init
3. escribes: git remote add origin [y la direccion del repo tuyo o del que vas a usar (sin los corchetes xd)]
4. si es tu primera vez solo escribes: git fetch origin main
5.  eso debería bastar para empezar y tener tu repo en local ya conectado con el remoto
6. en otra discusion por ahí vi que colocaron los 2 comandos para configurar de forma local el nombre y el correo, tambien en medio del proceso deberás ingresar por primer vez las credenciales que tienes en tu github para que le sea permitido trabajar.
7. VsCode tiene predefinido el uso de git, solo tienes que presionar archivo y "agregar carpeta al area de trabajo" (preferiblemente reiniciaes el Visual) y listo.
![image](https://user-images.githubusercontent.com/86509650/192933953-c583ac0d-5834-4a2a-9376-f5e3047de136.png)
8. en la pestaña de git puedes ver tu repo y para subir los cambios al remoto ya no tienes que usar codigo complicado.
![image](https://user-images.githubusercontent.com/86509650/192934126-554f685a-df50-4d10-a7be-1fc7f427571d.png)
te debe aparecer uno como esos.

**Comment by Amhedriel** [comment]
Buenas,

Daniel, te comparto un mini tutorial que tengo escrito en MarckDown [Tutoriales](https://github.com/Amhedriel/Mark_Down_Tutoriales "Tutorial en GitHub"), ahí podrás ver algunas cosas de Git, en especial la configuración para integrarlo en Visual Studio Code, que es el IDE que yo utilizo para Windows, que viendo el manual verás que es importante aclarar, si tienes alguna duda al respecto estoy a tu disposición.

Saludos.

**Comment by renzocastro** [comment]
En Youtube hay tienes variedad. Desde principiante a avanzado, de 5 min a 1 hora, Español o Inglés. Tú eliges cual encaja mejor para ti. https://www.youtube.com/results?search_query=git+tutorial
